### PR TITLE
Feature/unit tests

### PR DESCRIPTION
**Pull request by firethar**
Unit Tests

### CODE REVIEW

File: src/app/sold-products/sold-products.component.spec.ts [hunk 1]
```diff
@@ -1,14 +1,24 @@
...
 
   beforeEach(async () => {
     await TestBed.configureTestingModule({
-      declarations: [ SoldProductsComponent ]
+      imports: [
+        HttpClientModule
+      ],
+      declarations: [ SoldProductsComponent ],
+      schemas: [
+        CUSTOM_ELEMENTS_SCHEMA
```
Comment (michalzielonka) [inline src/app/sold-products/sold-products.component.spec.ts]: Why this schema?

File: src/app/app.component.spec.ts [hunk 2]
```diff
@@ -11,25 +15,24 @@ describe('AppComponent', () => {
       declarations: [
         AppComponent
       ],
+      schemas: [
+        CUSTOM_ELEMENTS_SCHEMA
```
Comment (michalzielonka) [inline src/app/app.component.spec.ts]: Why this schema?

File: src/app/sold-products/sold-products.component.spec.ts [hunk 3]
```diff
@@ -17,9 +27,29 @@ describe('SoldProductsComponent', () => {
...
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+ 
+  it('service should be injected', () => {
+    expect(service).toBeTruthy();
+  });
+
+  it('should get products$ Observable', (done) => {
```
Comment (michalzielonka) [inline src/app/sold-products/sold-products.component.spec.ts]: No need to test RxJS

File: src/app/sold-products/sold-products.component.spec.ts [hunk 4]
```diff
@@ -17,9 +27,29 @@ describe('SoldProductsComponent', () => {
...
+    component.products$.subscribe(value => {
+      expect(value).toBeTruthy();
+      done();
+    });
+  });
+
+  it('should get targetValue$ Observable', (done) => {
+    component.targetValue$.subscribe(value => {
```
Comment (michalzielonka) [inline src/app/sold-products/sold-products.component.spec.ts]: No need to test RxJS

**Review by michalzielonka — CHANGES_REQUESTED** [review]
Do not test RxJS in your code, test your logic. All the rest looks good. Also it is nice that you have found edge case issue. I was expecting that :)

File: src/app/sold-products/sold-products.component.spec.ts (hunk 1 above)
Comment (firethar) [reply src/app/sold-products/sold-products.component.spec.ts]: This was to prevent error:
`ERROR: 'NG0304: 'app-progress-bar' is not a known element:
1. If 'app-progress-bar' is an Angular component, then verify that it is part of this module.
2. If 'app-progress-bar' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.'`

But it seems that I have misunderstood this and I should simply declare ProgressBarComponent in foreach() block instead

File: src/app/app.component.spec.ts (hunk 2 above)
Comment (firethar) [reply src/app/app.component.spec.ts]: Same as above

File: src/app/sold-products/sold-products.component.spec.ts (hunk 4 above)
Comment (firethar) [reply src/app/sold-products/sold-products.component.spec.ts]: So how to cover this part of `ng test --no-watch --code-coverage`: 
image

File: src/app/sold-products/sold-products.component.spec.ts (hunk 1 above)
Comment (michalzielonka) [reply src/app/sold-products/sold-products.component.spec.ts]: Use NO_ERRORS_SCHEMA which is correct for this case

File: src/app/app.component.spec.ts (hunk 2 above)
Comment (michalzielonka) [reply src/app/app.component.spec.ts]: Same as previous comment

File: src/app/sold-products/sold-products.component.spec.ts (hunk 4 above)
Comment (michalzielonka) [reply src/app/sold-products/sold-products.component.spec.ts]: Coverage is not always the most important factor. If you want to check this check edge cases with some logic if possible. This way you only test RxJS map function which is tested in the framework.

**Review by michalzielonka — CHANGES_REQUESTED** [review]
Details in comments

File: src/app/sold-products/sold-products.component.spec.ts (hunk 1 above)
Comment (firethar) [reply src/app/sold-products/sold-products.component.spec.ts]: Done.

File: src/app/sold-products/sold-products.component.spec.ts (hunk 3 above)
Comment (firethar) [reply src/app/sold-products/sold-products.component.spec.ts]: Understood, test removed.

File: src/app/sold-products/sold-products.component.spec.ts (hunk 4 above)
Comment (firethar) [reply src/app/sold-products/sold-products.component.spec.ts]: Understood, test removed.

**Review by michalzielonka — APPROVED** [review]
